libs/kml/serdes_geojson.cpp

```diff
@@ -0,0 +1,219 @@
+
+  if (ec)
+  {
+    std::string err = glz::format_error(ec, json_content);
+    LOG(LWARNING, ("Error parsing JSON:", err));
+    return false;
+  }
+
+  // Copy bookmarks from parsed geoJsonData into m_fileData.
+  for (auto feature : geoJsonData.features)
+  {
+    if (feature.isPoint())
+    {
+      GeoJsonGeometryPoint const * point = std::get_if<GeoJsonGeometryPoint>(&feature.geometry);
+      double longitude = point->coordinates.at(0);
+      double latitude = point->coordinates.at(1);
+
+      std::map<std::string, glz::json_t> * props_json = &feature.properties;
+      BookmarkData bookmark;
+
+      // Parse label
+      auto nameVal = (*props_json)["name"];
+      auto labelVal = (*props_json)["label"];
+      if (nameVal.is_string())
+      {
+        auto name = kml::LocalizableString();
+        kml::SetDefaultStr(name, nameVal.get_string());
+        bookmark.m_name = name;
+      }
+      else if (labelVal.is_string())
+      {
+        auto name = kml::LocalizableString();
+        kml::SetDefaultStr(name, labelVal.get_string());
+        bookmark.m_name = name;
+      }
+
+      // Parse description
+      auto descriptionVal = (*props_json)["description"];
+      if (descriptionVal.is_string())
+      {
+        auto descr = kml::LocalizableString();
+        kml::SetDefaultStr(descr, descriptionVal.get_string());
+        bookmark.m_description = descr;
+      }
+
+      // Parse color
+      auto markerColorVal = (*props_json)["marker-color"];
+      if (markerColorVal.is_string())
+      {
+        auto colorRGBA = ParseColor(markerColorVal.get_string());
+        if (colorRGBA)
+          bookmark.m_color = ColorData{.m_rgba = *colorRGBA};
+      }
+
+      // Parse icon
+      // if (props_json->contains("marker-symbol") && (*props_json)["marker-symbol"].is_string()) {
+      //    auto const markerSymbol = (*props_json)["marker-symbol"].get<std::string>()
+      //    bookmark.m_icon = ???;
+      //}
+
+      // UMap custom properties
+      auto umapOptionsVal = (*props_json)["_umap_options"];
+      if (umapOptionsVal.is_object())
+      {
+        glz::json_t::object_t umap_options = umapOptionsVal.get_object();
+        // Parse color from properties['_umap_options']['color']
+        auto colorVal = umap_options["color"];
+        if (colorVal.is_string())
+        {
+          auto colorRGBA = ParseColor(colorVal.get_string());
+          if (colorRGBA)
+            bookmark.m_color = ColorData{.m_rgba = *colorRGBA};
+        }
+
+        // TODO: Store '_umap_options' as a JSON string in some bookmark field.
+      }
+
+      bookmark.m_point = mercator::FromLatLon(latitude, longitude);
+      m_fileData.m_bookmarksData.push_back(bookmark);
+    }
+    else if (feature.isUnknown())
+    {
+      GeoJsonGeometryUnknown const * unknownGeometry = std::get_if<GeoJsonGeometryUnknown>(&feature.geometry);
+      LOG(LWARNING, ("GeoJson contains unsupported geometry type:", unknownGeometry->type));
+    }
+  }
```

Comment:
If GeoJson contains a geometry which is not 'Point' and not 'LineString' then it's cocidered as unknown. Logging could help debug such cases.

Reply:
This would require a lot of changes in `libs/map/bookmark_manager.cpp` API. Currently we have only `success` flag there.